test/e2e/etcd_multi_node_test.go

```diff
@@ -0,0 +1,435 @@
+			Expect(deployBackupSecret(parentCtx, cl, logger, provider, etcdNamespace, storageContainer))
+
+		})
+
+		AfterEach(func() {
+			// remove etcd objects if any old etcd objects exists.
+			purgeEtcd(parentCtx, cl, providers)
+		})
+
+		It("should perform etcd operations", func() {
+			ctx, cancelFunc := context.WithTimeout(parentCtx, 10*time.Minute)
+			defer cancelFunc()
+
+			etcd := getDefaultMultiNodeEtcd(etcdName, namespace, storageContainer, storePrefix, provider)
+			objLogger := logger.WithValues("etcd-multi-node", client.ObjectKeyFromObject(etcd))
+
+			By("Create etcd")
+			createAndCheckEtcd(ctx, cl, objLogger, etcd)
+
+			By("Hibernate etcd (Scale down from 3->0)")
+			hibernateAndCheckEtcd(ctx, cl, objLogger, etcd)
+
+			By("Wakeup etcd (Scale up from 0->3)")
+			// scale up etcd replicas to 3 and ensures etcd cluster with 3 replicas is ready.
+			Expect(cl.Get(ctx, client.ObjectKeyFromObject(etcd), etcd)).To(Succeed())
+			etcd.Spec.Replicas = multiNodeEtcdReplicas
+			updateAndCheckEtcd(ctx, cl, logger, etcd)
+
+			By("Zero downtime rolling updates")
+			job := startEtcdZeroDownTimeValidatorJob(ctx, cl, etcd, "rolling-update")
+			// this defer ensures to remove the job, if test case breaks before deleting the job.
+			defer cleanUpTestHelperJob(ctx, cl, job.Name)
+
+			Expect(cl.Get(ctx, client.ObjectKeyFromObject(etcd), etcd)).To(Succeed())
+			// trigger rolling update by updating etcd quota
+			etcd.Spec.Etcd.Quota.Add(*resource.NewMilliQuantity(int64(10), resource.DecimalSI))
+			updateAndCheckEtcd(ctx, cl, objLogger, etcd)
+			checkEtcdZeroDownTimeValidatorJob(ctx, cl, client.ObjectKeyFromObject(job), objLogger)
+			cleanUpTestHelperJob(ctx, cl, job.Name) // remove job
+
+			By("Zero downtime maintenance operation: defragmentation")
+			job = startEtcdZeroDownTimeValidatorJob(ctx, cl, etcd, "defragmentation")
+			// this defer ensures to remove the job, if test case breaks before deleting the job.
+			defer cleanUpTestHelperJob(ctx, cl, job.Name)
+
+			objLogger.Info("Configure defragmentation schedule for every 1 minute")
+			Expect(cl.Get(ctx, client.ObjectKeyFromObject(etcd), etcd)).To(Succeed())
+			*etcd.Spec.Etcd.DefragmentationSchedule = "*/1 * * * *"
+			updateAndCheckEtcd(ctx, cl, objLogger, etcd)
+
+			checkDefragmentationFinished(ctx, cl, etcd, objLogger)
+
+			objLogger.Info("Checking any Etcd downtime")
+			// Checking Etcd cluster is healthy and there is no downtime while defragmentation.
+			// K8s job zeroDownTimeValidator will fail, if there is any downtime in Etcd cluster health.
+			checkEtcdZeroDownTimeValidatorJob(ctx, cl, client.ObjectKeyFromObject(job), objLogger)
+			cleanUpTestHelperJob(ctx, cl, job.Name)
+
+			By("Delete etcd")
+			deleteAndCheckEtcd(ctx, cl, objLogger, etcd)
+		})
+	})
+
+	Context("when single-node is configured", func() {
+		var (
+			etcdName         string
+			storageContainer string
+			parentCtx        context.Context
+			provider         TestProvider
+		)
```

Comment:
I guess there is no need to re-declare them if you declare them in inside `Describe(){...}`

Reply:
Fixed